### PR TITLE
Removed netcoreapp3.1, net5.0 and net6.0 + added net8.0 support

### PR DESCRIPTION
**Pull request by armanossiloko**
- Removed support for netcoreapp3.1 as it's officially out of support
- Removed support for net5.0
- Added net8.0 support
- Recreated the Middleware sample projects

### CODE REVIEW

**Comment by armanossiloko** [comment]
@khellang Did you plan on migrating your middleware repo to net8.0 and should I go on and close the PR if you prefer to keep things as they are?

File: src/ProblemDetails/ProblemDetails.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <Version>6.5.1</Version>
-    <TargetFrameworks>netcoreapp3.1;net5.0;net6.0</TargetFrameworks>
+    <Version>7.0.0</Version>
+    <TargetFrameworks>net6.0;net8.0</TargetFrameworks>
```
Comment (khellang) [inline src/ProblemDetails/ProblemDetails.csproj:4]: I think we can just remove .NET 6 as well. It's only a few weeks until it's out of support.

File: src/SpaFallback/SpaFallback.csproj [hunk 2]
```diff
@@ -14,7 +14,8 @@
     <PackageReference Include="Microsoft.Extensions.Options" Version="2.1.1" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
+  <ItemGroup Condition="'$(TargetFramework)' == 'net8.0'">
```
Comment (khellang) [inline src/SpaFallback/SpaFallback.csproj:17]: Is this necessary anymore?

File: test/Directory.Build.props [hunk 3]
```diff
@@ -3,16 +3,16 @@
     <OutputType>Exe</OutputType>
     <IsPackable>False</IsPackable>
     <LangVersion>9.0</LangVersion>
-    <TargetFrameworks>netcoreapp3.1;net6.0</TargetFrameworks>
+    <TargetFrameworks>net6.0;net8.0</TargetFrameworks>
```
Comment (khellang) [inline test/Directory.Build.props:6]: Same goes for these.

File: Hellang.Middleware.sln [hunk 4]
```diff
@@ -25,19 +25,17 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SpaFallback", "src\SpaFallb
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SpaFallback.Tests", "test\SpaFallback.Tests\SpaFallback.Tests.csproj", "{9EAA652B-873A-4F0F-B751-5381CDB774FD}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Authentication.JwtBearer.Google", "src\Authentication.JwtBearer.Google\Authentication.JwtBearer.Google.csproj", "{E5BCFA2D-ACE6-4BC6-A327-009563366065}"
```
Comment (khellang) [inline Hellang.Middleware.sln:28]: Why was this removed?

**Review by khellang — CHANGES_REQUESTED** [review]
Thanks for the contribution! 🙏🏻  Just a few nits from me 😄

File: Hellang.Middleware.sln (hunk 4 above)
Comment (armanossiloko) [reply Hellang.Middleware.sln]: I had it removed while doing the version changes and I thought I had re-added it, but I suppose not. It should be back in now.

File: test/Directory.Build.props (hunk 3 above)
Comment (armanossiloko) [reply test/Directory.Build.props]: Removed net6.0 support.

File: src/ProblemDetails/ProblemDetails.csproj (hunk 1 above)
Comment (armanossiloko) [reply src/ProblemDetails/ProblemDetails.csproj]: I initially kept it in just for these last few weeks, but I think you're right - net6.0 support has been removed.

File: src/ProblemDetails/ProblemDetails.csproj [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <Version>6.5.1</Version>
-    <TargetFrameworks>netcoreapp3.1;net5.0;net6.0</TargetFrameworks>
+    <Version>7.0.0</Version>
```
[the remark this answers is not in the document]
Comment (armanossiloko) [reply src/ProblemDetails/ProblemDetails.csproj:3]: Nice, thanks.

**Review by khellang — APPROVED** [review]

**Comment by khellang** [comment]
Looks like the build failed because GitHub Actions only has the `8.0.402` SDK. Can we just try to remove the `global.json` entirely and go with the default?

**Comment by armanossiloko** [comment]
> Looks like the build failed because GitHub Actions only has the `8.0.402` SDK. Can we just try to remove the `global.json` entirely and go with the default?

Absolutely. I've removed it now. Let's see how it goes.

**Comment by armanossiloko** [comment]
lol Looks like the build is good now, but the SpaFallback tests are failing - guess changing the target framework to net8.0 was enough to break its behavior. :D

**Review by khellang — APPROVED** [review]

**Comment by khellang** [comment]
Thanks for your awesome contribution, @armanossiloko! 🙏🏻

**Pull request merged**